compiler/src/dotty/tools/dotc/typer/ConstFold.scala

```diff
@@ -59,6 +59,16 @@ object ConstFold:
       tree.withFoldedType(Constant(targ.tpe))
     case _ => tree
 
+  private def treeConstant(tree: Tree)(using Context): Option[Constant] =
```

Comment:
I think it would be more elegant if `ConstantTree` was an extractor.

Reply:
Moved to `ConstantTree` extractor